src/main/java/org/jabref/gui/Main.css

```diff
@@ -66,6 +66,12 @@
     -fx-effect:none;
 }
 
+.search-tooltip {
```

Comment:
I agree that the default JavaFX tooltips are hard to read! But I'm not a fan of the old swing tooltips either. I would suggest to use a grayish color scheme with less opacity (>= 90%): [inspiration](https://material.io/guidelines/components/tooltips.html#tooltips-tooltips-desktop). Maybe increasing the text size (by default its 0.85em).

Please override the default `.tooltip` style so that tooltips look consistent.

Reply:
The problem is that the TextFlow is not styled together with the tooltip its inside. So only the background can be changed with the tooltip class when using a TextFlow. It would be needed to add a style class to each used texts inside the textflow i think? then i was able to change the text font. But this would mean it would be necessary to make a text style class for each different style: bold, italic, monospaced etc. In my tests the css style had overwritten the programatically set font.

So im a little confused what to do here. It seems difficult to make a standardized look here. Depending on what is used inside the tooltip: normal text <-> TextFlow

Even if i create all the different text style classes, they had to be used explicitly in all other places. So no automatic default style possible here i think? Or is there any other possible solution i dont see?

Reply:
You are talking about the font size? Using `-fx-font-size` does not work?
Moreover, you can influence the look of sub-items by using the appropriate css selector. For example, with `.tooltip > .text` you should be able to modify the look of the text elements in the tooltip. However, I have to admit that I don't really understand where you problem lies.

Reply:
The selector ```.tooltip > .text { ...``` does not work here. The text element has no default style class.  
I just found out I could use this: ```.tooltip Text { ...``` but then all texts have the same style. But we need to have some Texts with Bold, etc. properties and then the Text.setFont method for setting the font progammatically is not working anymore..

Reply:
That is strange. Usually styles specified in the code take priority over css files. You could add [PseudoClasses](https://docs.oracle.com/javase/8/javafx/api/javafx/css/PseudoClass.html) (e.g. `bold`) to the text element and style them via the selector `.tooltip Tex :bold`. 

But maybe the best solution is to actually leave all text related styles at their default value and only play a bit with the background color?

Reply:
That would be the easiest solution :) 

But a good styling would be nice too.

I played around a bit and the following would also be possible:  
```css
.tooltip Text {
    /* this styles all texts inside a tooltip */
    -fx-fill: rgba(255, 255, 255, 0.9);
    -fx-font-size: 1em;
}

.tooltip > TextFlow > .tooltip-text-bold {
    -fx-font-weight: bold;
}

.tooltip > TextFlow > .tooltip-text-italic {
    -fx-font-style: italic;
}

.tooltip > TextFlow > .tooltip-text-monospaced {
    -fx-font-family: monospace;
}
```
So every text gets a basic styling and when you want to make one Text e.g. bold you can add the ```"tooltip-text-bold"``` style class, then additionally the text will be bold. It is even possible to combine styles like italic and bold

I think I could integrate this rather easily. I didn't look into PseudoClasses but if you say it is the best way to go I can look into them. So what should it be?

Reply:
Your solution looks very clean! Go for it, if it works! (Pseudo classes are very similar, not sure what is preferable here).